timon/plugins/http_server/views.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+# #############################################################################
+# Copyright : (C) 2023 by MHComm. All rights reserved
+#
+# __author__ = "Quentin Laymajoux"
+# __email__ = "info@mhcomm.fr"
+#
+# Name       : timon.plugins.http_server.views
+"""
+Summary: The http server plugin views and routes
+"""
+# #############################################################################
+import logging
+
+from quart_trio import QuartTrio
+
+from timon.probes.probe_if import mk_probe
+
+logger = logging.getLogger(__name__)
+
+app = QuartTrio(__name__)
+
+KNOWN_ROUTES = {
+    "/resources/": "returns a list of all resources and their availability",
+    "/heap/": "returns the lenght of the heap",
+    "/probes/<?probename>/run/": ("force run the probename and returns "
+                                  "the result"),
+}
+
+
+@app.route("/")
+async def get_index():
+    """
+    returns a list of all routes and help text
+    """
+    return KNOWN_ROUTES
+
+
+@app.route("/resources/")
+async def get_resources():
+    """
+    returns a list of all resources and their availability
+    """
+    rsrc_infos = {}
+    rsrcs = app.tmoncfg.queue.get_resources()
+    for rsrc_name, rsrc in rsrcs.items():
+        semaph = rsrc.semaph
+        rsrc_infos[rsrc_name] = {
+            "value": semaph.value,
+        }
+    return rsrc_infos
+
+
+@app.route("/heap/")
+async def get_heap_len():
+    """
+    returns the lenght of the heap
+    """
```

Comment:
I also wouldn't call it heap.

If you think, 'heap' is an implementation detail.

What this is in reality is the probe_queue or list of scheduled_probes 
But doesn't it always have the same length, which is the amount of all existing probes?

or perhaps it changes a little (but not a lot)
amount of all existing probes 'minus' amount if currently running probes.

as normally a probe is dequeed just to be run and to be re-enqueued

Reply:
waiting_probes / scheduled_probes ??

Reply:
Question !! "queue" instead of "heap" will be good ?

Reply:
yes